src/h5web/visualizations/line/hooks.ts

```diff
@@ -0,0 +1,18 @@
+import type ndarray from 'ndarray';
+import { useMemo } from 'react';
+import { ScaleType, Domain } from '../shared/models';
+import { findDomain, extendDomain } from '../shared/utils';
+
+export function useAxisDomain(
+  dataArray: ndarray,
+  scaleType: ScaleType
+): Domain | undefined {
+  return useMemo(() => {
+    const isLogScale = scaleType === ScaleType.Log;
+    const rawDomain = isLogScale
+      ? findDomain((dataArray.data as number[]).filter((x) => x > 0))
+      : findDomain(dataArray.data as number[]);
```

Comment:
For readability, I would store `dataArray.data` in a constant and move the ternary inside `findDomain`.